### PR TITLE
Add documentation for `zarr.core.group`

### LINKED ISSUE #2246: Add documentation for `zarr.group`
xref: #2215

This issue tracks the documentation progress for the [`zarr.core.group`](https://github.com/zarr-developers/zarr-python/blob/v3/src/zarr/core/group.py) module. We'll be focusing on adding docstrings and code samples to the functions.

Progress:

- [ ] docstrings
- [ ] code samples

### PR DESCRIPTION
**Pull request by brokkoli71**
- fixes #2246

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Review by jhamman — COMMENTED** [review]
Great start here @brokkoli71. The doc build is failing. Can you take a look at that and then we can do a final review?

**Comment by brokkoli71** [comment]
feel free to give feedback if you think documentation is missing or that there is unnecessary information.

File: src/zarr/core/group.py [hunk 1]
```diff
@@ -1348,6 +1567,29 @@ def from_store(
...
+
+        Returns
+        -------
+        Group
+            Group instantiated from the store.
+
+        Raises
+        ------
+        ContainsArrayError, ContainsGroupError, ContainsArrayAndGroupError
+
```
Comment (jhamman) [inline src/zarr/core/group.py:1591]:
```suggestion
```

File: src/zarr/core/group.py [hunk 2]
```diff
@@ -1366,10 +1608,50 @@ def open(
...
+        Returns
+        -------
+        Array | Group
+            Group member (Array or Group) at the specified key
+
+        Examples
+        --------
+        >>> import zarr
+        >>> group = Group.from_store(zarr.storage.MemoryStore(mode="w"))
```
Comment (jhamman) [inline src/zarr/core/group.py:1644]:
```suggestion
        >>> group = Group.from_store(zarr.storage.MemoryStore()
```

File: src/zarr/core/group.py [hunk 3]
```diff
@@ -1390,32 +1672,104 @@ def get(self, path: str, default: DefaultT | None = None) -> Array | Group | Def
         -------
         object
             Group member (Array or Group) or default if not found.
+
+        Examples
+        --------
+        >>> import zarr
+        >>> group = Group.from_store(zarr.storage.MemoryStore(mode="w"))
```
Comment (jhamman) [inline src/zarr/core/group.py:1679]:
```suggestion
        >>> group = Group.from_store(zarr.storage.MemoryStore()
```

File: src/zarr/core/group.py [hunk 4]
```diff
@@ -1390,32 +1672,104 @@ def get(self, path: str, default: DefaultT | None = None) -> Array | Group | Def
...
+
+        Parameters
+        ----------
+        key : str
+            Group member name.
+
+        Examples
+        --------
+        >>> import zarr
+        >>> group = Group.from_store(zarr.storage.MemoryStore(mode="w"))
```
Comment (jhamman) [inline src/zarr/core/group.py:1705]:
```suggestion
        >>> group = Group.from_store(zarr.storage.MemoryStore()
```

**Review by jhamman — APPROVED** [review]
A few small edits but this looks great. Thanks @brokkoli71!

**Comment by jhamman** [comment]
@brokkoli71 - there are some conflicts here but I'm ready to merge once those are resolved.

**Pull request merged**